### ISSUE TITLE
[1.4.0.1] Shell crashes when using tab completion on require call

### ISSUE THREAD

**Issue by lieuwex** [meteor/meteor#7623]
I'm running Meteor version 1.4.0.1 on Mac OS X 10.11.6

On a new project, when using `meteor shell` and running:
```
> require('meteor/http').<tab>
```

, the Meteor server crashes with:
```
W20160810-15:47:34.353(2)? (STDERR) readline.js:925
W20160810-15:47:34.401(2)? (STDERR)             throw err;
W20160810-15:47:34.401(2)? (STDERR)             ^
W20160810-15:47:34.401(2)? (STDERR)
W20160810-15:47:34.401(2)? (STDERR) TypeError: Cannot convert undefined or null to object
W20160810-15:47:34.401(2)? (STDERR)     at keys (native)
W20160810-15:47:34.401(2)? (STDERR)     at Function.keys (/Users/lieuwe/.meteor/packages/es5-shim/.4.6.13.1pc6ile++os+web.browser+web.cordova/npm/node_modules/es5-shim/es5-shim.js:1084:20)
W20160810-15:47:34.402(2)? (STDERR)     at REPLServer.self._domain.on.REPLServer.complete (repl.js:673:25)
W20160810-15:47:34.402(2)? (STDERR)     at REPLServer.complete [as completer] (repl.js:329:10)
W20160810-15:47:34.402(2)? (STDERR)     at REPLServer.Interface._tabComplete (readline.js:375:8)
W20160810-15:47:34.402(2)? (STDERR)     at REPLServer.Interface._ttyWrite (readline.js:872:16)
W20160810-15:47:34.402(2)? (STDERR)     at PassThrough.onkeypress (readline.js:106:10)
W20160810-15:47:34.402(2)? (STDERR)     at emitTwo (events.js:87:13)
W20160810-15:47:34.402(2)? (STDERR)     at PassThrough.emit (events.js:172:7)
W20160810-15:47:34.402(2)? (STDERR)     at emitKeys (readline.js:1252:14)
=> Exited with code: 1
```

It doesn't matter if the package being required exists or not.

I tried to find the error but I'm not really sure where the completions are even provided, I also am not sure if this is a problem with Meteor or Node.

It kind of reminded me of nodejs/node#7716 but it seems like different errors.

**Comment by benjamn** [comment]
This appears to be because `require.extensions` (a Node-specific CommonJS extension) is [not defined](https://github.com/nodejs/node/blob/0974fc6a25c343744235331eb20d31f6412ff7e1/lib/repl.js#L673). Since the implementation of `meteor shell` is part of the command-line tool, a proper fix will have to wait until Meteor 1.4.1: https://github.com/meteor/meteor/pull/7624

Note: while tab completion definitely shouldn't crash the server, you probably shouldn't expect `require("meteor/http").<tab>` to show any meaningful completions, since `require("meteor/http")` is a function call that might have side effects or return different values each time you call it, so it's not safe for the shell to call it repeatedly.

**Comment by benjamn** [comment]
This is fixed in Meteor 1.4.1, available now via `meteor update`. Feel free to reopen this issue if the problem persists.

**Comment by lieuwex** [comment]
Thanks! 😊
